File: VB/App_Code/MyDataViewTemplate.vb
#Region "#templateUsings"
Imports DevExpress.Web
Imports System.Web.UI
#End Region ' #templateUsings

Public Class MyDataViewTemplate
    Implements ITemplate

    #Region "#template"
    Public Sub InstantiateIn(ByVal container As Control) Implements ITemplate.InstantiateIn
        Dim dataViewContainer = TryCast(container, DataViewItemTemplateContainer)
        container.Controls.Add(GetFormLayout(dataViewContainer.DataItem))
    End Sub
    Private Function GetFormLayout(ByVal dataItem As Object) As ASPxFormLayout
        Dim formLayout = New ASPxFormLayout()
        formLayout.ID = "FormLayout"
        Dim layoutGroup = TryCast(formLayout.Items.Add(New LayoutGroup("Item Info")), LayoutGroup)
        Dim layoutItemName = New LayoutItem("Name")
        layoutItemName.Controls.Add(New ASPxLabel() With {.Text = GetData(dataItem, "Name")})
        Dim layoutItemDecsription = New LayoutItem("Description")
        layoutItemDecsription.Controls.Add(New ASPxLabel() With {.Text = GetData(dataItem, "Description")})
        layoutGroup.Items.Add(layoutItemName)
        layoutGroup.Items.Add(layoutItemDecsription)
        Return formLayout
    End Function
    Private Function GetData(ByVal dataObject As Object, ByVal fieldName As String) As String
        Return DataBinder.Eval(dataObject, fieldName).ToString()
    End Function
    #End Region ' #template
End Class

File: CS/App_Code/MyDataViewTemplate.cs
#region #templateUsings
using DevExpress.Web;
using System.Web.UI;
#endregion #templateUsings

public class MyDataViewTemplate : ITemplate {
    #region #template
    public void InstantiateIn(Control container) {
        var dataViewContainer = container as DataViewItemTemplateContainer;
        container.Controls.Add(GetFormLayout(dataViewContainer.DataItem));
    }
    ASPxFormLayout GetFormLayout(object dataItem) {
        var formLayout = new ASPxFormLayout();
        formLayout.ID = "FormLayout";
        var layoutGroup = formLayout.Items.Add(new LayoutGroup("Item Info")) as LayoutGroup;
        var layoutItemName = new LayoutItem("Name");
        layoutItemName.Controls.Add(new ASPxLabel() { Text = GetData(dataItem, "Name") });
        var layoutItemDecsription = new LayoutItem("Description");
        layoutItemDecsription.Controls.Add(new ASPxLabel() { Text = GetData(dataItem, "Description") });
        layoutGroup.Items.Add(layoutItemName);
        layoutGroup.Items.Add(layoutItemDecsription);
        return formLayout;
    }
    string GetData(object dataObject, string fieldName) {
        return DataBinder.Eval(dataObject, fieldName).ToString();
    }
    #endregion #template
}